processing/src/main/java/io/druid/query/extraction/ExplicitDimRenameFn.java

```diff
@@ -0,0 +1,119 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.extraction;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Function;
+import com.google.common.base.Strings;
+import com.google.common.base.Throwables;
+import io.druid.jackson.DefaultObjectMapper;
+
+import javax.annotation.Nullable;
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+/**
+ *
+ */
+public class ExplicitDimRenameFn extends DimExtractionFn
+{
+  @JsonIgnore
+  private final ObjectMapper mapper;
+
+  private final Map<String, String> renames;
+  private final Function<String, String> extractionFunction;
+  private final ExtractionType extractionType;
+
+  @JsonCreator
+  public ExplicitDimRenameFn(
+      @JsonProperty("renames") final Map<String, String> renames,
+      @JsonProperty("uniqueRenames") final Boolean uniqueRenames
+  )
+  {
+    this.mapper = new DefaultObjectMapper();
+    this.renames = renames;
+    this.extractionFunction = new Function<String, String>()
+    {
+      @Nullable
+      @Override
+      public String apply(@Nullable String input)
+      {
+        final String retval = renames.get(input);
+        return Strings.isNullOrEmpty(retval) ? input : retval;
+      }
+    };
+    this.extractionType = (uniqueRenames != null && uniqueRenames) ? ExtractionType.RENAME : ExtractionType.REBUCKET;
+  }
+
+  private static final byte CACHE_TYPE_ID = 0x5;
+
+  @JsonProperty("renames")
+  public Map<String, String> getRenames()
+  {
+    return renames;
+  }
+
+  @JsonProperty("uniqueRenames")
+  public Boolean getUniqueRenames()
+  {
+    return ExtractionType.RENAME.equals(getExtractionType());
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    final byte[] bytes;
+    try {
+      bytes = mapper.writeValueAsBytes(renames);
+    }
+    catch (JsonProcessingException e) {
+      // Should never happen. If it can't map a Map<String, String> then there's something weird
+      throw Throwables.propagate(e);
+    }
+
+    return ByteBuffer
+        .allocate(bytes.length + 1)
+        .put(CACHE_TYPE_ID)
+        .put(bytes)
+        .array();
+  }
+
+  @Override
+  public String apply(String value)
+  {
+    return this.extractionFunction.apply(value);
```

Comment:
Yes, this is personal taste, but looking through other parts of the code I don't see it very often so I'll remove it. Mostly because I tend to be writing constructors or setters which contain `this.somePassedField = somePassedField` so my brain has equated `this.someField` with the way to indicate a class field's value.